### PR TITLE
Move timing component working copy code into a mixin. Add input valid…

### PR DESCRIPTION
**Pull request by wongvincent**
…ation prior to closing dialog.

Invalid input error detection code taken from the `d2l-activity-editor-container-mixin`: https://github.com/BrightspaceHypermediaComponents/activities/blob/651c0f5e1f46be0fd1b1ee1ec5aa818cf516976c/components/d2l-activity-editor/mixins/d2l-activity-editor-container-mixin.js#L183

---------------------------------------------
https://rally1.rallydev.com/#/29180338367d/iterationstatus?detail=%2Fuserstory%2F478606416460%2Ftasks

Thoughts?

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-timing-container.js [hunk 1]
```diff
@@ -40,44 +30,20 @@ class ActivityQuizManageTimingContainer extends ActivityEditorMixin(ActivityEdit
...
 				id="quiz-manage-timing-dialog"
 				?opened="${this.opened}"
-				@d2l-dialog-close="${this._closeDialog}"
-				width=800
+				@d2l-dialog-close="${this.closeDialog}"
+				width=900
 				title-text=${this.localize('subHdrTimingTools') }>
 					<div id="manage-timing-dialog-timing-editor">${this._renderQuizTimingEditor()}</div>
-					<d2l-button slot="footer" primary data-dialog-action="ok">${this.localize('manageTimingDialogConfirmationText')}</d2l-button>
+					<d2l-button slot="footer" primary @click="${this._checkinDialog}">${this.localize('manageTimingDialogConfirmationText')}</d2l-button>
```
Comment (emil-furniss) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-timing-container.js:46]: Call `this.checkinDialog` directly?

File: components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	constructor(store) {
+		super();
+		this.checkoutOnLoad = true;
+		this.dialogHref = '';
+		this.store = store;
+	}
+
+	async checkinDialog(e, skipValidatingInputs) {
```
Comment (emil-furniss) [inline components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js:21]: What's the use-case for skipping the validation?

File: components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	async save() {
+		const entity = this.store.get(this.checkedOutHref);
+		if (!entity) return;
+
+		await entity.checkin(this.store);
+	}
+
+	async _focusOnInvalid() {
```
Comment (emil-furniss) [inline components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js:57]: If this is being re-used, maybe we can move them to a `util` or something like that?

File: components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		return {
+			dialogHref: { type: String }
+		};
+	}
+
+	constructor(store) {
+		super();
+		this.checkoutOnLoad = true;
+		this.dialogHref = '';
+		this.store = store;
```
Comment (emil-furniss) [inline components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js:18]: Could you call `super(store)` and then have the `ActivityEditorMixin` provide `this.store`?

**Comment by mostlyfabulous** [comment]
Was chatting with Ronan yesterday about this behaviour being extracted into a `mixin` but was told you were already on it 🚀 
Will help a lot with all the other quiz properties we need to working copy-ify.

File: components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js (hunk 2 above)
Comment (wongvincent) [reply components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js:22]: Good question, nothing, so I'll remove supporting skipping validation.

File: components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js (hunk 4 above)
Comment (wongvincent) [reply components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js:19]: yessir

**Review by ronan-f — APPROVED** [review]
LGTM

File: components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js (hunk 3 above)
Comment (wongvincent) [reply components/d2l-activity-editor/mixins/d2l-activity-editor-working-copy-dialog-mixin.js:56]: I'm still looking into this, but I will merge this for now and refactor in a different PR.

**Pull request merged**